storage/innobase/fil/fil0fil.cc

```diff
@@ -7498,34 +7529,39 @@ fil_encryption_rotate()
 			continue;
 		}
 
-		if (space->encryption_type != Encryption::NONE) {
-			mtr_start(&mtr);
+		bool success = fil_encryption_rotate_low(space);
+		if (!success) {
+			return(false);
+		}
 
-			if (fsp_is_system_temporary(space->id)) {
-				mtr_set_log_mode(&mtr, MTR_LOG_NO_REDO);
-			}
+		space = UT_LIST_GET_NEXT(space_list, space);
+		DBUG_EXECUTE_IF("ib_crash_during_rotation_for_encryption",
```

Comment:
No testcase is added using this - is this an omission (then please add it) or is it a backport of Oracle patch with presumably non-public testcase - in which case don't add this code?

Reply:
I didn't add this. The only change here is refactoring into fil_encryption_rotate_low() and yes this looks like a dbug crash point used by Oracle's non-public testcase

Reply:
So it needs removing then?

Reply:
No, previously (before my change) it is ..
fil_encryption_rotate() {
  for () {
     ..
    if(space->encryption != Encryption::None) {
      ..
     }
     ..
  }
  DBUG_EXECUTE_IF("ib_crash_during_rotation_for_encryption",
                                  DBUG_SUICIDE(););
}

Now:
fil_encryption_rotate() {
  for () {
     ..
    fil_encryption_rotate_low(space);
     ..
  }
  DBUG_EXECUTE_IF("ib_crash_during_rotation_for_encryption",
                                  DBUG_SUICIDE(););
}

I dont see the need to remove this crash point. Only the code above was refactored into separate function.

Reply:
Indeed, sorry, OK to leave as is